tests/e2e/Services/Projects/ProjectsConsoleClientTest.php

```diff
@@ -3793,4 +3793,273 @@ public function testDeleteProject(): array
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()), [
+            'key' => str_repeat("A", 256),
+            'value' => 'TESTINGVALUE'
+        ]);
+
+        $this->assertEquals(400, $variable['headers']['status-code']);
+
+        // Test for invalid value
+        $variable = $this->client->call(Client::METHOD_POST, '/project/variables', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()), [
+            'key' => 'LONGKEY',
+            'value' => str_repeat("#", 8193),
+        ]);
+
+        $this->assertEquals(400, $variable['headers']['status-code']);
+
+        return array_merge(
+            $data,
+            [
+                'variableId' => $variableId,
+            ]
+        );
+    }
+
+    /**
+     * @depends testCreateProjectVariable
+     */
+    public function testListVariables(array $data)
+    {
+        /**
+         * Test for SUCCESS
+         */
+
+        $response = $this->client->call(Client::METHOD_GET, '/project/variables', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertEquals(1, sizeof($response['body']['variables']));
+        $this->assertEquals(1, $response['body']['total']);
+        $this->assertEquals("APP_TEST", $response['body']['variables'][0]['key']);
+        $this->assertEquals("TESTINGVALUE", $response['body']['variables'][0]['value']);
+
+        return $data;
+    }
+
+    /**
+     * @depends testListVariables
+     */
+    public function testGetVariable(array $data)
+    {
+        /**
+         * Test for SUCCESS
+         */
+        $response = $this->client->call(Client::METHOD_GET, '/project/variables/' . $data['variableId'], array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()));
+
+        $this->assertEquals(200, $response['headers']['status-code']);
+        $this->assertEquals("APP_TEST", $response['body']['key']);
+        $this->assertEquals("TESTINGVALUE", $response['body']['value']);
+
+        /**
+         * Test for FAILURE
+         */
+
+        $response = $this->client->call(Client::METHOD_GET, '/project/variables/NON_EXISTING_VARIABLE', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $data['projectId'],
+            'x-appwrite-mode' => 'admin',
+        ], $this->getHeaders()));
+
+        $this->assertEquals(404, $response['headers']['status-code']);
+
+        return $data;
+    }
+
+    /**
+     * @depends testGetVariable
```

Comment:
We can mark this as depending on testCreateVariable, mainly for clarity reasons. Logic-wise its fine both ways